cogito/protocol_test.go

```diff
@@ -71,10 +65,23 @@ func TestSourceValidationFailure(t *testing.T) {
 
 	testCases := []testCase{
 		{
-			name:    "missing mandatory source keys",
+			name:    "missing mandatory git source keys",
 			source:  cogito.Source{},
 			wantErr: "source: missing keys: owner, repo, access_token",
 		},
+		{
+			name:    "missing mandatory gchat source keys",
+			source:  cogito.Source{Sinks: []string{"gchat"}},
+			wantErr: "source: missing keys: gchat_webhook",
+		},
+		{
+			name: "invalid sink source keys",
+			source: cogito.Source{
+				Sinks:        []string{"gchat", "slacks"},
+				GChatWebHook: "sensitive-gchat-webhook",
```

Comment:
why do you need this here and not before?

Reply:
Yes, just a few lines above, you don't mark anything as sensitive. In example L74 still uses Gchat but you don't configure `GChatWebHook`. Why is it needed in this test?

Reply:
Ah I understand now.
If you want to use the `gchat` feature of Cogito, `GChatWebHook`  becomes a mandatory parameter. In L74, we test just  that failure: `"source: missing keys: gchat_webhook"`.
Because Source validation comes before anything else, I need to specify `GChatWebHook` in the other places to test other error messages.  
About `sensitive-gchat-webhook` I'll change if it confuses, you can put what you want actually.